File: 3447-clear-digits/solution.py
class Solution:
    def clearDigits(self, s: str) -> str:
        list1=list(s)
        i=0
        while i<len(list1):
            if list1[i].isdigit():
                list1.pop(i)
                if i-1>=0:
                    list1.pop(i-1)
                    i-=1
            else:
                i+=1
        return ''.join(list1)

        

File: 3447-clear-digits/solution.java
class Solution {
    public String clearDigits(String s) {
        Stack<Character> st = new Stack<>();
        for(char i:s.toCharArray()){
            if(i>='0' && i<='9'){
                if(!st.empty())
                    st.pop();
            }
            else{
                st.push(i);
            }
        }
        String res = "";
        while(!st.empty())
            res=st.pop()+res;
        return res;
    }
}
